File: Java/Polonez.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        int woman = 0, man = 0;
        Scanner input = new Scanner(System.in);
        input.nextLine();
        String[] class3a = input.nextLine().split(" ");
        String[] class3b = input.nextLine().split(" ");
        String[] class4j = input.nextLine().split(" ");
        for (String name : class3a) 
        {
            if (name.endsWith("a")) 
                woman++;
            else 
                man++;
        }
        for (String name : class3b) 
        {
            if (name.endsWith("a")) 
                woman++;
            else 
                man++;
        }
        for (String name : class4j) 
        {
            if (name.endsWith("a")) 
                woman++;
            else 
                man++;
        }
        System.out.println(Math.min(woman, man));
        input.close();
    }
}

File: C++/Polonez.cpp
#include <vector>
#include <string>
#include <iostream>
#include <algorithm>

int main() 
{
    int a, b, c, woman = 0, man = 0;
    std::cin >> a >> b >> c;
    std::vector<std::string> class3a(a), class3b(b), class4j(c);
    for (std::string &name : class3a) 
        std::cin >> name;
    for (std::string &name : class3b) 
        std::cin >> name;
    for (std::string &name : class4j) 
        std::cin >> name;
    auto count_gender = [&](const std::vector<std::string> &names)
    {
        for (const std::string &name : names) 
        {
            if (name.back() == 'a') 
                woman++;
            else 
                man++;
        }
    };
    count_gender(class3a);
    count_gender(class3b);
    count_gender(class4j);
    std::cout << std::min(woman, man) << '\n';
    return 0;
}